### PR TITLE
Version compatibility fixes

### PR DESCRIPTION
**Pull request by jcrist**
Updates for NumPy 1.10 and Pandas 0.17. Also pin a few package versions for python2.6 to speed up conda solve time.

### CODE REVIEW

File: dask/dataframe/tests/test_io.py [hunk 1]
```diff
@@ -295,7 +295,7 @@ def test_compression_multiple_files():
 
 def test_empty_csv_file():
     with filetext('a,b') as fn:
-        df = dd.read_csv(fn)
+        df = dd.read_csv(fn, header=0)
```
Comment (mrocklin) [inline dask/dataframe/tests/test_io.py]: What happened here?

**Comment by mrocklin** [comment]
Looks good to me, curious to see what travis times are.

File: dask/dataframe/tests/test_io.py (hunk 1 above)
Comment (jcrist) [inline dask/dataframe/tests/test_io.py]: This was a bug that was masked by pandas implicitly casting. The test was actually asserting dask has different behavior than pandas - this fixes it so the two remain compatible.

**Comment by mrocklin** [comment]
It looks like py33 is also slow.  Should the special conda line apply to 3.3 as well?

**Comment by jcrist** [comment]
Normally it's just as fast as the other runs - not sure why it was slow here.

**Comment by jcrist** [comment]
I'm going to restart that build, and see if it's slow again.

**Comment by jcrist** [comment]
Odd. Look at this build from earlier today: https://travis-ci.org/blaze/dask/builds/84969701. 3.3 was just as fast as the others. Now it's slow. Did something change in the last few hours?  Or is this just intermediate?

**Comment by mrocklin** [comment]
In principle I would expect 3.3 to suffer the same issues that 2.6 does.
Neither has recent builds through anaconda.

**Comment by jcrist** [comment]
Pinned 3.3 as well. All tests pass, merging.

**Pull request merged**